packages/nimble-components/src/anchor-tabs/index.ts

```diff
@@ -0,0 +1,295 @@
+            subtree: true,
+            attributeFilter: ['aria-disabled']
+        });
+    }
+
+    private readonly isDisabledElement = (el: Element): el is HTMLElement => {
+        return el.getAttribute('aria-disabled') === 'true';
+    };
+
+    private readonly isFocusableElement = (el: Element): el is HTMLElement => {
+        return !this.isDisabledElement(el);
+    };
+
+    private readonly setTabs = (): void => {
+        const gridHorizontalProperty = 'gridColumn';
+        const gridVerticalProperty = 'gridRow';
+
+        this.activetab = undefined;
+        this.tabs.forEach((tab: HTMLElement, index: number) => {
+            const tabId: string = this.tabIds[index]!;
+            const isActiveTab = this.activeid === tabId && this.isFocusableElement(tab);
+            tab.setAttribute('id', tabId);
+            tab.setAttribute('aria-selected', isActiveTab ? 'true' : 'false');
+            tab.addEventListener('click', this.handleTabClick);
+            tab.addEventListener('keydown', this.handleTabKeyDown);
+            tab.setAttribute('tabindex', isActiveTab ? '0' : '-1');
+            if (isActiveTab) {
+                this.activetab = tab;
+            }
+
+            tab.style[gridVerticalProperty] = '';
+            tab.style[gridHorizontalProperty] = `${index + 1}`;
+        });
+    };
+
+    private getTabIds(): string[] {
+        return this.tabs.map((tab: HTMLElement) => {
+            return tab.getAttribute('id') ?? `tab-${uniqueId()}`;
+        });
+    }
+
+    private readonly handleTabClick = (event: MouseEvent): void => {
+        const selectedTab = event.currentTarget as HTMLElement;
+        if (
+            selectedTab.nodeType === 1
+            && this.isFocusableElement(selectedTab)
+        ) {
+            this.tabs.forEach((tab: HTMLElement) => {
+                tab.setAttribute('tabindex', tab === selectedTab ? '0' : '-1');
+            });
+            this.navigateToTab(
+                this.tabs.findIndex((item: HTMLElement) => item === selectedTab)
+            );
+        }
+    };
+
+    private readonly handleTabKeyDown = (event: KeyboardEvent): void => {
+        let anchor;
+        switch (event.key) {
+            case keyArrowLeft:
+                event.preventDefault();
+                this.adjustBackward();
+                break;
+            case keyArrowRight:
+                event.preventDefault();
+                this.adjustForward();
+                break;
+            case keyHome:
+                event.preventDefault();
+                this.focusFirstOrLast(false);
+                break;
+            case keyEnd:
+                event.preventDefault();
+                this.focusFirstOrLast(true);
+                break;
+            case keySpace:
+            case keyEnter:
+                event.preventDefault();
+                event.target?.dispatchEvent(new Event('click'));
+                break;
+            case 'ContextMenu':
```

Comment:
Wow, is this code handling a keyboard key that invokes a context menu? Somebody thought of everything--we're even handling cases for keys that don't exist in fast-utilities!

Reply:
Yep, that's what it's for. I hope there aren't any other one-offs that we missed.